x-pack/plugins/security/server/config_deprecations.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ConfigDeprecationProvider } from 'src/core/server';
```

Comment:
__optional nit:__  (feel free to skip it if you're not going to merge master and wait for CI once again)
```suggestion
import type { ConfigDeprecationProvider } from 'src/core/server';
```